server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -144,9 +144,10 @@ private static TransportVersion registerTransportVersion(int id, String uniqueId
     public static final TransportVersion V_8_500_017 = registerTransportVersion(8_500_017, "0EDCB5BA-049C-443C-8AB1-5FA58FB996FB");
     public static final TransportVersion V_8_500_018 = registerTransportVersion(8_500_018, "827C32CE-33D9-4AC3-A773-8FB768F59EAF");
     public static final TransportVersion V_8_500_019 = registerTransportVersion(8_500_019, "09bae57f-cab8-423c-aab3-c9778509ffe3");
+    public static final TransportVersion V_8_500_020 = registerTransportVersion(8_500_020, "803e260d-9c0b-42a2-aa59-58e2b2024285");
```

Comment:
FYI, this is needed for the stateless PR. The new getCommitExtraUserData() alone do not need a new transport version. Might be confusing for whoever in the future may try to find why this new transport version was introduced and may look at the commit/PR. I was definitely confused in the beginning of reading this PR why a new transport version was needed.

Do you think it'd make sense to do the new transport version in a separate PR? Or at least in the final commit description make a not that the transport version is unrelated to the new getCommitExtraUserData().